library/src/main/java/org/xmtp/android/library/Conversations.kt

```diff
@@ -433,74 +452,209 @@ data class Conversations(
         return hmacKeysResponse.build()
     }
 
-    private suspend fun listIntroductionPeers(pagination: Pagination? = null): Map<String, Date> {
-        val apiClient = client.apiClient ?: throw XMTPException("V2 only function")
-        val envelopes = apiClient.queryTopic(
-            topic = Topic.userIntro(client.address),
-            pagination = pagination,
-        ).envelopesList
-        val messages = envelopes.mapNotNull { envelope ->
-            try {
-                val message = MessageV1Builder.buildFromBytes(envelope.message.toByteArray())
-                // Attempt to decrypt, just to make sure we can
-                message.decrypt(client.privateKeyBundleV1)
-                message
-            } catch (e: Exception) {
-                Log.d(TAG, e.message.toString())
-                null
-            }
+    private suspend fun handleConsentProof(
+        consentProof: Invitation.ConsentProofPayload,
+        peerAddress: String,
+    ) {
+        val signature = consentProof.signature
+        val timestamp = consentProof.timestamp
+
+        if (!KeyUtil.validateConsentSignature(signature, client.address, peerAddress, timestamp)) {
```

Comment:
Do we need to log an error/warning, and possibly return it here?

Reply:
I don't think this code changed at all in this PR. Something to visit whenever we touch consent proofs for V3